nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureDataLakeStorage.java

```diff
@@ -76,27 +109,45 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             final DataLakeServiceClient storageClient = getStorageClient(context, flowFile);
             final DataLakeFileSystemClient fileSystemClient = storageClient.getFileSystemClient(fileSystem);
             final DataLakeDirectoryClient directoryClient = fileSystemClient.getDirectoryClient(directory);
-            final DataLakeFileClient fileClient = directoryClient.createFile(fileName);
+            final DataLakeFileClient fileClient;
+
+            final String conflictResolution = context.getProperty(CONFLICT_RESOLUTION).getValue();
+            boolean overwrite = conflictResolution.equals(REPLACE_RESOLUTION);
+
+            try {
+                fileClient = directoryClient.createFile(fileName, overwrite);
+
+                final long length = flowFile.getSize();
+                if (length > 0) {
+                    try (final InputStream rawIn = session.read(flowFile); final BufferedInputStream in = new BufferedInputStream(rawIn)) {
+                        fileClient.append(in, 0, length);
+                    }
+                }
+                fileClient.flush(length);
+
+                final Map<String, String> attributes = new HashMap<>();
+                attributes.put("azure.filesystem", fileSystem);
+                attributes.put("azure.directory", directory);
+                attributes.put("azure.filename", fileName);
+                attributes.put("azure.primaryUri", fileClient.getFileUrl());
+                attributes.put("azure.length", String.valueOf(length));
+                flowFile = session.putAllAttributes(flowFile, attributes);
 
-            final long length = flowFile.getSize();
-            if (length > 0) {
-                try (final InputStream rawIn = session.read(flowFile); final BufferedInputStream in = new BufferedInputStream(rawIn)) {
-                    fileClient.append(in, 0, length);
+                session.transfer(flowFile, REL_SUCCESS);
+                final long transferMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
+                session.getProvenanceReporter().send(flowFile, fileClient.getFileUrl(), transferMillis);
+            } catch (DataLakeStorageException dlsException) {
+                if (dlsException.getStatusCode() == 409) {
+                    if (conflictResolution.equals(IGNORE_RESOLUTION)) {
+                        session.transfer(flowFile, REL_SUCCESS);
+                        getLogger().warn("Transferring {} to success because file with same name already exists", new Object[]{flowFile});
```

Comment:
The warning message does not properly describe the cause and the effect: file exists => transfer to success
The reason for transferring to success is the 'Ignore' resolution policy rather.
It should also be mentioned that the file has not been overwritten in Azure.